block/manager.go

```diff
@@ -816,20 +816,21 @@ func (m *Manager) recordMetrics(block *types.Block) {
 func (m *Manager) submitBlocksToDA(ctx context.Context) error {
 	submitted := false
 	backoff := initialBackoff
-	blocks := m.pendingBlocks.getPendingBlocks()
+	blocksToSubmit := m.pendingBlocks.getPendingBlocks()
 	for attempt := 1; ctx.Err() == nil && !submitted && attempt <= maxSubmitAttempts; attempt++ {
-		res := m.dalc.SubmitBlocks(ctx, blocks)
+		res := m.dalc.SubmitBlocks(ctx, blocksToSubmit)
 		switch res.Code {
 		case da.StatusSuccess:
 			m.logger.Info("successfully submitted Rollkit block to DA layer", "daHeight", res.DAHeight, "count", res.SubmittedCount)
-			if int(res.SubmittedCount) == len(blocks) {
+			if int(res.SubmittedCount) == len(blocksToSubmit) {
 				submitted = true
```

Comment:
I think we could just remove this and update the `if !submitted` check to `if len(m.pendingBlocks.getPendingBlocks()) == 0`

And the error message should be `failed to submit all blocks, submitted x of y blocks`
Then we can track the starting pending blocks, and the ending pending blocks for a more informative error message.

Reply:
then `submitted` should be `submittedAll` to be more accurate.